Show hide filters interaction [DAH-568]

james-wills-sf opened the pull request:
[DAH-568]
Also closes [DAH-161]

includes pattern library changes from https://github.com/SFDigitalServices/sf-dahlia-pattern-library/pull/159

![showhidefilters2](https://user-images.githubusercontent.com/64036574/98747002-7bc65380-236b-11eb-8449-b5c80b04b06d.gif)

![- 2020-11-10 at 3 45 41 PM](https://user-images.githubusercontent.com/64036574/98747178-df508100-236b-11eb-8ceb-ffb5bedea675.png)


[DAH-568]: https://sfgovdt.jira.com/browse/DAH-568

[DAH-161]: https://sfgovdt.jira.com/browse/DAH-161

james-wills-sf committed:
add show/hide filter functionality

james-wills-sf committed:
Add tests

akegan commented on app/javascript/components/atoms/colors.js:
```diff
@@ -34,5 +34,6 @@ export const COLORS = {
   warn: '#fdb81e',
-  warnTint: '#fee8b6'
+  warnTint: '#fee8b6',
+  whiteSmoke: '#f5f5f5'
```
is whitesmoke one that I missed from pattern lib or a new color?

james-wills-sf replied:
@akegan it existed in the pattern library, I think it wasn't carried over somehow

software-project reviewed (approved):
It looks good to me. Only case where filter state might be a little misleading is when you change the value of a select and change it back to a previous one. Apply button is blue but technically filters are not changed. But I don't think it's a big deal. All in all nice work 👍

akegan reviewed (changes requested):
Looks good, one thing is that the apply and clear are supposed to be aligned right I think.

Should we fix up styling for the search in this PR too or wait until we do bulk actions? Looks like the input needs to be a little taller unfortunately (or the buttons a bit smaller) and maybe the search button is supposed to have the blue/gray behavior instead of being disabled when there isn't a change? It also looks like the search button and the show/hide filters button have different heights.

Also wondering when we should add the labels for the filter fields, maybe when we switch to multi-selects? Just want to make sure that those changes don't fall through the cracks of the tickets.

james-wills-sf commented:
@akegan I'll take a stab at the styling. Question about your labels comment, I thought I added labels for the filters? Or maybe I'm not sure what you're talking about

akegan commented:
I just totally missed them 🤦. I was just thinking about the "Any accessibility request" option that was still in there as being a substitute for the label, but we just still need that null label until we do the multi-select.

james-wills-sf committed:
add examples for extra padding button

james-wills-sf committed:
fix styling

james-wills-sf commented:
@akegan okay styling issues should be fixed. I went ahead and switched "disabled" with tertiary-inverse on the search button because that's how it is in the designs.


ONly thing I didn't do anything about is: "Looks good, one thing is that the apply and clear are supposed to be aligned right I think."

I think they are right aligned?

Review app:
<img width="1083" alt="- 2020-11-12 at 4 39 29 PM" src="https://user-images.githubusercontent.com/64036574/99013612-b9f77a80-2505-11eb-83a7-e9c288900961.png">

[Figma designs](https://www.figma.com/file/p3ir6cmCeZwqYNVCoqTpPb/applicant-table-DAH-104?node-id=286%3A3020):
<img width="540" alt="- 2020-11-12 at 4 39 51 PM" src="https://user-images.githubusercontent.com/64036574/99013622-bebc2e80-2505-11eb-8ed7-ba2d26faf054.png">

akegan commented:
OK this one was real! They are left aligned for me still, maybe a firefox/chrome difference?
![image](https://user-images.githubusercontent.com/11825994/99084646-384f2d80-2595-11eb-9d42-019b58a66c97.png)

Other changes look good except now the clear icon isn't centered. It's absolutely positioned so the margin is hardcoded 
![image](https://user-images.githubusercontent.com/11825994/99084719-4dc45780-2595-11eb-93e9-415d22651922.png)

just need to change [this variable](https://github.com/SFDigitalServices/sf-dahlia-pattern-library/blob/main/public/toolkit/styles/molecules/_inputs.scss#L389) to 17px (input is now 50px high total, (50-16)/2 = 17px.)

james-wills-sf committed:
address comments

james-wills-sf commented:
@akegan okay I addressed your comments

The alignment issue was because I used `justifyContent: 'end'` instead of `justifyContent: 'flex-end'`. I looked for a linting rule that would catch this in the future but couldn't find one.

I also fixed the search padding by adding to this pattern library pr: https://github.com/SFDigitalServices/sf-dahlia-pattern-library/pull/160

akegan reviewed (approved):
looks good now. thanks for the changes
